docs/api/java/providers.md

```diff
@@ -0,0 +1,372 @@
+}
+```
+
+#### Inputs and outputs
+
+| Name    | Description                            |
+|---------|----------------------------------------|
+| from    | Offset of tokens.                      |
+| list    | Limit of amount of tokens to return.   |
+| returns | Prepared get confirmed tokens request. |
+
+### `zksGetBridgeContracts`
+
+Get default bridges addresses deployed on L1 and L2.
+
+Example:
+
+```java
+import io.zksync.protocol.ZkSync;
+import org.web3j.protocol.http.HttpService;
+import io.zksync.protocol.core.BridgeAddresses;
+
+public class Main { 
+    public static void main(String ...args) {
+      BridgeAddresses bridgeAddresses = zksync.zksGetBridgeContracts().sendAsync().join().getResult();
+}
+```
+
+#### Inputs and outputs
+
+| Name    | Description                           |
+|---------|---------------------------------------|
+| from    | Offset of tokens.                     |
+| list    | Limit of amount of tokens to return.  |
+| returns | Prepared get bridge contract request. |
+
+### `zksMainContract`
+
+Get address of main contract for current network chain.
+
+Example:
+
+```java
+import io.zksync.protocol.ZkSync;
+import org.web3j.protocol.http.HttpService;
+
+public class Main { 
+    public static void main(String ...args) {
+      String mainContract = zksync.zksMainContract().sendAsync().join().getResult();
+}
+```
+
+#### Inputs and outputs
+
+| Name    | Description                     |
+|---------|---------------------------------|
+| returns | Prepared main contract request. |
+
+### `zksGetTestnetPaymaster`
+
+Get the address of the testnet paymaster: the paymaster that is available on testnets and enables paying fees in ERC-20 compatible tokens.
+
+Example:
+
+```java
+import io.zksync.protocol.ZkSync;
+import org.web3j.protocol.http.HttpService;
+import io.zksync.methods.response.ZksTestnetPaymasterAddress;
+
+public class Main { 
+    public static void main(String ...args) {
+      ZksTestnetPaymasterAddress response = zksync.zksGetTestnetPaymaster().send();
+}
+```
+
+#### Inputs and outputs
+
+| Name    | Description                     |
+|---------|---------------------------------|
+| returns | Prepared get paymaster request. |
+
+### `zksGetTokenPrice`
+
+Get price of the token in USD.
+
+Example:
+
+```java
+import io.zksync.protocol.ZkSync;
+import org.web3j.protocol.http.HttpService;
+
+public class Main { 
+    public static void main(String ...args) {
+      ZksTokenPrice response = zksync.zksGetTokenPrice(ETH.getL2Address()).send();
+}
+```
+
+#### Inputs and outputs
+
+| Name         | Description                        |
+|--------------|------------------------------------|
+| tokenAddress | Address of the token in hex forma. |
+| returns      | Prepared get token price request.  |
+
+### `zksGetTransactionReceipt`
+
+Get transaction receipt. The same as eth_getTransactionReceipt but with additional fields belong to L2ToL1 messenger
```

Comment:
```suggestion
Get transaction receipt. The same as `eth_getTransactionReceipt` but with additional fields belong to L2ToL1 messenger
```

Reply:
Should L2ToL1 messenger be L2ToL1Messenger ??

Reply:
It should be L2ToL1Log